gateway/src/shard/processor.rs

```diff
@@ -50,14 +55,34 @@ impl ShardProcessor {
             properties,
             rx,
             session: Arc::new(Session::new(tx)),
+            inflater: Inflater::new(),
         })
     }
 
     pub async fn run(mut self) {
         let mut remove_listeners = Vec::new();
 
         loop {
-            let gateway_event = self.next_event().await;
+            let gateway_event = match self.next_event().await {
+                Ok(ev) => ev,
+                // Reconnect as this error is often fatal!
+                Err(Error::Decompressing {
+                    source,
+                }) => {
+                    warn!(
+                        "[gateway] Decompressing error, clears buffers and reconnect! {:?}",
+                        source
+                    );
+
+                    // Inflater gets reset in the reconnect call.
+                    self.reconnect().await;
+                    continue;
+                },
+                Err(err) => {
+                    warn!("Decompressing failed: {:?}", err.source());
```

Comment:
This warning isn't correct because this will be anything *but* a decompressing error, that's handled above